var/spack/repos/builtin/packages/py-scikit-odes/package.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyScikitOdes(PythonPackage):
+    """Odes is a scikit toolkit for scipy to add extra ode solvers.
+    Specifically it interfaces the Sundials solvers cvode, cvodes, ida and
+    idas.  It this way it provides extra modern ode and dae solvers you can
+    use, extending the capabilities offered in scipy.integrade.ode."""
+
+    homepage = "https://github.com/bmcage/odes"
+
+    pypi = "scikits.odes/scikits.odes-2.7.0.tar.gz"
+    github = "https://github.com/bmcage/odes.git"
+
+    maintainers("omsai")
+
+    version("2.7.0", sha256="a71e19e1485893754ae8c050668232fcc694f17b83602e75fbebf7bf9f975e1e")
+
+    depends_on("python@3.7:", type=("build", "run"))
+
+    depends_on("py-setuptools@:64.0.0", type="build")
+
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-cython@:3.0.0a7", type=("build", "run"))
```

Comment:
This package does not build with `py-cython@3.0.0`, as stated here correctly. As far as I know, spack sorts versions unintuitively sometimes. If I remember correctly spack sorts `py-cython@3.0.0` < `py-cython@3.0.0.a7`. For this specific case, this did not happen for me. The package was concretized to use `py-cython@0.29.36`. Ping @adamjstewart for his input.

Reply:
@manuelakuhn To me, your concretization is working correctly - the spec says < @3.0.0a7 and @3.0.0a7 < @3.0.0 beause the alpha version is less than the corresponding release version which you can also see in the pypi package release history: https://pypi.org/project/Cython/#history so the next best version in spack is 0.29.36 because the cython project jumped from that to version 3 for whatever reason.

Reply:
What I meant was that the sorting in spack is sometimes not as you would expect and I was told that spack would sort the alpha version as newer as the release version (so exactly the opposite to the normal way around). So the question was more to @adamjstewart if we have to add any precautions to prevent it. But since it seems to work fine for this specific case, I think we can just leave it as is.

Reply:
I see what you're saying: spack wrongly considers the alpha versions to be more recent than the release:

```console
$ spack python
Spack version 0.21.0.dev0
Python 3.10.12, Linux x86_64
>>> from spack.version import Version
>>> Version("3.0.0a7") > Version("3.0.0")
True
>>> Version("3.0.0.a7") > Version("3.0.0")
True 
```

I'll keep my eyes open about bugs relating to that and see how other packages with pre-3.0 cython dependencies manage their versioning.